### ISSUE TITLE
Turns on and off by it self

### ISSUE THREAD

**Issue by mkomarac** [arendst/Sonoff-Tasmota#2192]
Make sure these boxes are checked before submitting your issue - Thank you!

- Sonoff 4CH PRO
- Firmware 5.12.0

some channels are turning on and off without user interaction. 
I have several Sonoff devices flashed with Tasmota firmware. Sonoff Basic flashed with Tasmota works couple of months without single failure. After this very good experience I have bought two Sonoff 4ch PRO devices and strange things started to happened. Switches are turned on and off by them selfs, without any user interaction. It happens on both PRO's flashed with firmware 5.9.1. I have upgraded firmware to 5.12.0 and problems keeps going. 

Can someone help with this issue?
Does anyone else have similar problem?

I have posted same issue in Home Assistant forum.
https://community.home-assistant.io/t/tasmota-problem/47488
But I believe this is the place to post specific issue.

Thank You all in advance!
BR

**Comment by fpapale** [comment]
I have the same problem with a Sonoff Dual. Another same device works right; the firmware is the same: 5.12 for both devices

**Comment by fpapale** [comment]
I have the same problem with a Sonoff Dual. Another device works right; the firmware is the same: 5.12 for both devices

**Comment by mkomarac** [comment]
it's so irritating lights turning on late at night while watching tv, or going to sleep. 
And every time, light turns ON, manually turning off. 100 times. it drives me crazy. 
I don't know what to do, and I really love Tasmota, but it's not working reliably. I don't want to throw it away, but if the issue is unsolvable I need to revert to old school switches, and I don't want to do that.

**Comment by vMeph** [comment]
Have same problem somethimes

**Comment by arendst** [comment]
Mine don't

**Comment by mkomarac** [comment]
the PianSom on HA forum suggested this solution... 
https://www.youtube.com/watch?v=aq8_os6g13s
I'll try this and report back. 
but issue is also with 2nd device, on workbench without wires soldered...
I'll try to clear 433 MHz memory from PRO's to see will this help...
but somehow I don't believe this suggestions will solve the issue.

**Comment by wasi54** [comment]
The Problem is most likely the WiFi
Check if there is no metall around your sonoffs. If the sonoffs are closer than 2" to a metall plate, they will lose the wifi connection, You may try out with sonoffs already working and then put a metall plate beside the sonoff. Also make sure your wify signal is ok  and you do not have other wifis from your neighbor on the same channel. Set the wifi mode to retry instead to restart!
If you have wires soldered, additionally give a chance to the solution from mkomarac/PianSom on youtube.
I have more then 10 sonoffs (without wires soldered) running with 5.12.0 without any issues. Most of the sonoffs are up and running for more than 15 day's.

**Comment by mkomarac** [comment]
PRO's are mounted on metal rail!
and cabinet door are also metal!
I was suspicious about wifi signal... this is from Sonoff build into cabinet: AP1 SSId (RSSI) | WiFi (80%)

another PRO is on wooden workbench without metal plates around, and no wires soldered, and having similar issues... 

WiFi mode retry instead restart, is option on Sonoff or Router?

**Comment by wasi54** [comment]
Retry is a sonoff mode, see https://github.com/arendst/Sonoff-Tasmota/wiki/Commands
(WifiConfig) or modify user_config.h before compile to
#define WIFI_CONFIG_TOOL       WIFI_RETRY

**Comment by wasi54** [comment]
Try to leave the cabinet door open for some time to test !?

**Comment by wasi54** [comment]
And a wifi signal of 80% is not a guaranty. Reflection can also be an issue on strong signals.

**Comment by mkomarac** [comment]
> Retry is a sonoff mode, see https://github.com/arendst/Sonoff-Tasmota/wiki/Commands
(WifiConfig) or modify user_config.h before compile to
#define WIFI_CONFIG_TOOL WIFI_RETRY

Will You be so kind, and write complete command, so I can paste it into command line in web interface of Sonoff. I haven't compile it on my own with Arduino... I have flashed it with mirko/SONOTA method, so I haven't soldered jumper pins for FTDI adapter.

thanks

**Comment by mkomarac** [comment]
![image](https://user-images.githubusercontent.com/37035814/37593245-940d1112-2b71-11e8-87ec-5e62f7ad89fd.png)
take a look of number of turning on (and I manually each time turn off) today from 06:00 till now

**Comment by wasi54** [comment]
http://yousonoff/cm?cmnd=WifiConfig%204

**Comment by mkomarac** [comment]
thanks.
after apply this, the issue remains.

**Comment by wasi54** [comment]
Did you checked the console output? Debug?

**Comment by mkomarac** [comment]
this is from console:

13:35:38 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:35:38 MQT: stat/sonoff4_1/POWER4 = OFF
13:36:11 MQT: stat/sonoff4_1/RESULT = {"POWER1":"ON"}
13:36:11 MQT: stat/sonoff4_1/POWER1 = ON
13:36:14 MQT: stat/sonoff4_1/RESULT = {"POWER1":"OFF"}
13:36:14 MQT: stat/sonoff4_1/POWER1 = OFF
13:36:32 MQT: stat/sonoff4_1/RESULT = {"POWER1":"ON"}
13:36:32 MQT: stat/sonoff4_1/POWER1 = ON
13:36:34 MQT: stat/sonoff4_1/RESULT = {"POWER1":"OFF"}
13:36:34 MQT: stat/sonoff4_1/POWER1 = OFF
13:36:54 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T13:36:54","Uptime":"0T00:40:16","Vcc":3.150,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","Wifi":{"AP":1,"SSId":"WiFi","RSSI":82,"APMac":"F0:B4:29:D8:CC:6C"}}
13:37:27 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:37:27 MQT: stat/sonoff4_1/POWER4 = ON
13:37:53 MQT: Attempting connection...
13:37:53 MQT: Connected
13:37:53 MQT: tele/sonoff4_1/LWT = Online (retained)
13:37:53 MQT: cmnd/sonoff4_1/POWER = 
13:39:56 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:39:57 MQT: stat/sonoff4_1/POWER4 = OFF
13:41:04 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:41:04 MQT: stat/sonoff4_1/POWER4 = ON
13:41:11 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:41:11 MQT: stat/sonoff4_1/POWER4 = OFF
13:41:47 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:41:47 MQT: stat/sonoff4_1/POWER4 = ON
13:42:05 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T13:42:04","Uptime":"0T00:45:16","Vcc":3.151,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"ON","Wifi":{"AP":1,"SSId":"WiFi","RSSI":80,"APMac":"F0:B4:29:D8:CC:6C"}}
13:42:34 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:42:34 MQT: stat/sonoff4_1/POWER4 = OFF
13:42:35 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:42:35 MQT: stat/sonoff4_1/POWER4 = ON
13:42:36 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:42:36 MQT: stat/sonoff4_1/POWER4 = OFF
13:43:01 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:43:02 MQT: stat/sonoff4_1/POWER4 = ON
13:43:08 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:43:09 MQT: stat/sonoff4_1/POWER4 = OFF
13:45:24 MQT: stat/sonoff4_1/RESULT = {"POWER1":"ON"}
13:45:24 MQT: stat/sonoff4_1/POWER1 = ON
13:45:26 MQT: stat/sonoff4_1/RESULT = {"POWER2":"ON"}
13:45:26 MQT: stat/sonoff4_1/POWER2 = ON
13:45:27 MQT: stat/sonoff4_1/RESULT = {"POWER1":"OFF"}
13:45:27 MQT: stat/sonoff4_1/POWER1 = OFF
13:45:28 MQT: stat/sonoff4_1/RESULT = {"POWER2":"OFF"}
13:45:28 MQT: stat/sonoff4_1/POWER2 = OFF
13:45:43 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:45:43 MQT: stat/sonoff4_1/POWER4 = ON
13:45:46 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:45:46 MQT: stat/sonoff4_1/POWER4 = OFF
13:46:00 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:46:00 MQT: stat/sonoff4_1/POWER4 = ON
13:46:01 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:46:01 MQT: stat/sonoff4_1/POWER4 = OFF
13:46:45 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:46:45 MQT: stat/sonoff4_1/POWER4 = ON
13:47:10 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T13:47:10","Uptime":"0T00:50:16","Vcc":3.149,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"ON","Wifi":{"AP":1,"SSId":"WiFi","RSSI":80,"APMac":"F0:B4:29:D8:CC:6C"}}
13:47:57 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:47:57 MQT: stat/sonoff4_1/POWER4 = OFF
13:49:45 MQT: Attempting connection...
13:49:45 MQT: Connected
13:49:45 MQT: tele/sonoff4_1/LWT = Online (retained)
13:49:45 MQT: cmnd/sonoff4_1/POWER = 
13:51:24 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:51:24 MQT: stat/sonoff4_1/POWER4 = ON
13:52:19 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T13:52:19","Uptime":"0T00:55:16","Vcc":3.149,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"ON","Wifi":{"AP":1,"SSId":"WiFi","RSSI":80,"APMac":"F0:B4:29:D8:CC:6C"}}
13:53:30 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
13:53:30 MQT: stat/sonoff4_1/POWER4 = OFF
13:55:32 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
13:55:32 MQT: stat/sonoff4_1/POWER4 = ON

**Comment by mkomarac** [comment]
yeah
it's so funny sitting in the living room, watching tv, and light is coming on by itself, and I keep pressing off button.

it reminds me on The Big Bang Theory when some Chinese guys across the World have turned lights @ Lenards home 😄 
same thing is happening to me, my wife is so angry... daughter says it wasn't me... 😄 
Wife Approval Factor is not going to pass if this keeps happening 😃

**Comment by wasi54** [comment]
Console output with more debug?

**Comment by mkomarac** [comment]
15:06:54 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:06:54 MQT: stat/sonoff4_1/POWER4 = OFF
15:07:22 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:07:22 MQT: stat/sonoff4_1/POWER4 = ON
15:07:31 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:07:31 MQT: stat/sonoff4_1/POWER4 = OFF
15:08:55 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:08:55 MQT: stat/sonoff4_1/POWER4 = ON
15:09:17 MQT: Attempting connection...
15:09:17 MQT: Connected
15:09:17 MQT: tele/sonoff4_1/LWT = Online (retained)
15:09:17 MQT: cmnd/sonoff4_1/POWER = 
15:09:37 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T15:09:37","Uptime":"0T02:10:16","Vcc":3.148,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"ON","Wifi":{"AP":1,"SSId":"WiFi","RSSI":82,"APMac":"F0:B4:29:D8:CC:6C"}}
15:09:59 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:09:59 MQT: stat/sonoff4_1/POWER4 = OFF
15:10:28 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:10:28 MQT: stat/sonoff4_1/POWER4 = ON
15:11:09 MQT: Attempting connection...
15:11:09 MQT: Connected
15:11:09 MQT: tele/sonoff4_1/LWT = Online (retained)
15:11:09 MQT: cmnd/sonoff4_1/POWER = 
15:11:13 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:11:13 MQT: stat/sonoff4_1/POWER4 = OFF
15:13:03 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:13:03 MQT: stat/sonoff4_1/POWER4 = ON
15:13:24 MQT: Attempting connection...
15:13:24 MQT: Connected
15:13:24 MQT: tele/sonoff4_1/LWT = Online (retained)
15:13:24 MQT: cmnd/sonoff4_1/POWER = 
15:13:46 MQT: Attempting connection...
15:13:46 MQT: Connected
15:13:46 MQT: tele/sonoff4_1/LWT = Online (retained)
15:13:46 MQT: cmnd/sonoff4_1/POWER = 
15:14:55 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T15:14:55","Uptime":"0T02:15:16","Vcc":3.149,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"ON","Wifi":{"AP":1,"SSId":"WiFi","RSSI":80,"APMac":"F0:B4:29:D8:CC:6C"}}
15:15:32 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:15:32 MQT: stat/sonoff4_1/POWER4 = OFF
15:16:40 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:16:40 MQT: stat/sonoff4_1/POWER4 = ON
15:17:28 RSL: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:17:33 RSL: stat/sonoff4_1/POWER4 = OFF
15:17:38 MQT: Attempting connection...
15:17:38 MQT: Connected
15:17:38 MQT: tele/sonoff4_1/LWT = Online (retained)
15:17:38 MQT: cmnd/sonoff4_1/POWER = 
15:18:13 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:18:13 MQT: stat/sonoff4_1/POWER4 = ON
15:18:44 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:18:44 MQT: stat/sonoff4_1/POWER4 = OFF
15:20:13 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T15:20:13","Uptime":"0T02:20:16","Vcc":3.150,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","Wifi":{"AP":1,"SSId":"WiFi","RSSI":76,"APMac":"F0:B4:29:D8:CC:6C"}}
15:20:17 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:20:17 MQT: stat/sonoff4_1/POWER4 = ON
15:21:45 MQT: Attempting connection...
15:21:45 MQT: Connected
15:21:45 MQT: tele/sonoff4_1/LWT = Online (retained)
15:21:46 MQT: cmnd/sonoff4_1/POWER = 
15:22:19 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:22:19 MQT: stat/sonoff4_1/POWER4 = OFF
15:23:23 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:23:23 MQT: stat/sonoff4_1/POWER4 = ON
15:25:21 MQT: tele/sonoff4_1/STATE = {"Time":"2018-03-19T15:25:21","Uptime":"0T02:25:16","Vcc":3.148,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"ON","Wifi":{"AP":1,"SSId":"WiFi","RSSI":80,"APMac":"F0:B4:29:D8:CC:6C"}}
15:26:23 MQT: Attempting connection...
15:26:23 MQT: Connected
15:26:23 MQT: tele/sonoff4_1/LWT = Online (retained)
15:26:23 MQT: cmnd/sonoff4_1/POWER = 
15:26:38 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:26:38 MQT: stat/sonoff4_1/POWER4 = OFF
15:27:00 MQT: stat/sonoff4_1/RESULT = {"POWER4":"ON"}
15:27:00 MQT: stat/sonoff4_1/POWER4 = ON
15:27:10 MQT: stat/sonoff4_1/RESULT = {"POWER4":"OFF"}
15:27:10 MQT: stat/sonoff4_1/POWER4 = OFF

**Comment by arendst** [comment]
As it's power4 most of the times I suggest, as I did before in other issues, you to clear the RF signals on your Sonoff Pro. They have nothing to do with Tasmota as it's an independent circuit controlled by the microcontroller receiving RF signals and controlling the relays.

Search for clearing RF signals on iTead website and then return here if it still persists.

**Comment by arendst** [comment]
As another item it could also be a retained power4 toggle message as you use Hass and I hate Hass as it uses retained messages people just don't understand the consequences of it.

**Comment by mkomarac** [comment]
sure, I'll clear RF signals, and report back.
As I want to eliminate 2nd sonoff PRO from interfering, I pressed Reset Configuration in Tasmota menu, and can't access tasmota anymore over network, and not even can't connect to it's own Hot Spot to configure network. it seems it's bricked. And it worked yesterday...
I have flashed Tasmota to original ITEAD software with [mirko/SONOTA](https://github.com/mirko/SonOTA) method. And it worked great. How to flash Tasmota again to unresponsive Sonoff device?

**Comment by Jason2866** [comment]
Have you tried the long press button reset method?
https://github.com/arendst/Sonoff-Tasmota/wiki/Button-usage

**Comment by mkomarac** [comment]
@arendst 
I have cleared RF codes by long press button to clear RF codes, but it haven't solved the problem...
it did, partially, but now the problem has reflected to switch 3 :(
now, after cleared RF codes, (test time 30 min, yeah, I know, it is too short time to tell anything), but now switch 4 remains off, but switch 3... I select switch 3 to ON mode, and after 8 mins it switches to OFF without any user interaction, just by it self. 
So, one more thing remains to try, is RC filter...
I'll try that as soon as I came home during daytime... and have time to dismantle complete installation...
and of course get back here...
so please, don't close this thread. I hope someone will have use of it!


You were referring to HA and retaining messages.
Would You be so kind, and give more detailed explanation what You mean about it, hating HA, and retaining messages?

I can say, that Home Assistant is exactly what I have looking for whole my life. Affordable, simple and nice way to have smart home. 

thanks

**Comment by mkomarac** [comment]
after couple of days testing, it seems that problem is related with MQTT broker. 
any ideas?

**Comment by Jason2866** [comment]
See #2140

**Comment by mkomarac** [comment]
After lot's of testing, the problem is with manual buttons.

if I push button on the device, (or other push button dislocated away from PRO by STP cable) it won’t retain state, and turn ON/OFF; to be exactly different state, than it was.
but, if I control PRO’s by “soft” switches, HA Interface, HADashboard, or Alexa. everything is working as it should. So maybe problem lies by controlling PRO (or Tasmota) by hard wire switches!?
push button is connected to GPIOx in PRO’s and it triggers relay. I don’t see difference activating relays by pushing hard button, or any other soft button.

**Comment by Jason2866** [comment]
PullUp resistors installed?

**Comment by mkomarac** [comment]
NO.

**Comment by myfingerhurt** [comment]
You may try to add 1K Pull-Up resistors to your GPIOx, might be helpful.

**Comment by schumi2004** [comment]
Same issue here with a Basic version after 1 day of perfect working.

Log says this:
2018-06-15 19:12:40.364 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":1,"svalue":"","Battery":55,"RSSI":5}
2018-06-15 19:12:40.482 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":0,"svalue":"","Battery":55,"RSSI":4}
2018-06-15 19:13:07.788 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":1,"svalue":"","Battery":55,"RSSI":5}
2018-06-15 19:13:07.911 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":0,"svalue":"","Battery":55,"RSSI":5}
2018-06-15 19:13:19.407 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":1,"svalue":"","Battery":55,"RSSI":5}
2018-06-15 19:13:19.531 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":0,"svalue":"","Battery":55,"RSSI":5}
2018-06-15 19:14:20.119 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":1,"svalue":"","Battery":55,"RSSI":5}
2018-06-15 19:14:20.239 MQTT: Topic: domoticz/in, Message: {"idx":23,"nvalue":0,"svalue":"","Battery":55,"RSSI":5}

Going to try the Pull-up resistor thing.
During my investigation i disabled all communication with Domitcz in anyway and it still happend, after reboot of the Sonoff Basic ot only got worse.

**Comment by HobbytronicsPK** [comment]
I'm having the same issue with Sonoff Slampher running Tasmota 6.1.0. 
I am not even using MQTT and disabled it under settings.
Slampher turns light ON at 5:20 AM in the morning automatically without any user interaction. It also turns light ON several times a day.

**Comment by ciprian288** [comment]
Same issue  with Sonoff 4CH pro Tasmota 6.1.0. When i set 2 releys on inching mode and restart my sonoff the inching releys  are turning on . I am not using MQTT .

**Comment by jasonehines** [comment]
I have this issue with Sonoff Basic, unmodified (except for Tasmota), with version 6.1.1. I've even tried resetting and reflashing it via the GPIO.

I also have it on Basics that have with switches attached to GPIO 14. They would occasionally turn on by themselves on version 5.14 but it's much worse wit 6.1.1.

**Comment by HobbytronicsPK** [comment]
I have experienced ghost switching if I forget to erase flash memory before
flashing Tasmota firmware in my Sonoff devices. Erasing flash memory and
re-flashing the Tasmota resolved the issue in my case.

So always erase the Flash memory before flashing new Sonoff Tasmota
Firmware. Have a look on to the following guide for more information.

https://hobbytronics.pk/sonoff-original-firmware-backup-restore/#Erase-Flash-Memory

**Comment by Jason2866** [comment]
Do you use in your broker "retain"? If yes thats your issue.
If not, do you have connected switches via wires? 
If yes disconnect for testing the wires, and see if ghost switchimg is gone.

**Comment by jasonehines** [comment]
@Jason2866 I have explicitly set retain to False. The Sonoff is stock, except for the firmware.

**Comment by HobbytronicsPK** [comment]
Jason, have you turned the MQTT OFF in Tasmota?

Have a look on to
https://hobbytronics.pk/yeti-tasmota-control-sonoff-without-internet/ for
controlling Sonoff without MQTT and Raspberry Pi Server using yeti smart
phone app.

**Comment by HobbytronicsPK** [comment]
It is a complete setup discussed on the link
https://hobbytronics.pk/yeti-tasmota-control-sonoff-without-internet/.
Please have a look on to it for more information. For a brief overview, it
uses Sonoff Tasmota and Yeti, a smart phone app to control Sonoff devices
directly.

Regards,



*Waqas Ahmed*



*Hobbytronics PK*

Mochiwali, Kahror Pakka

59340, Lodhran, Punjab PK

*Call:* +(92 - 03) 007 3210 41

*Web:* www.hobbytronics.pk

**Comment by mkomarac** [comment]
After a while without ghost switching, here we go again!
The Issue is in Sonoff 4ch PRO 
it is powered by 12V DC and switch 3 and 4 are wired with extension STP cables to additional switches
this is switch 3
![image](https://user-images.githubusercontent.com/37035814/43514223-00bdfbe6-9580-11e8-916d-ebcdfdeed425.png)
and this is switch 4
![image](https://user-images.githubusercontent.com/37035814/43514276-263c9742-9580-11e8-85f5-03c3ac87287c.png)

Do You think that pull up resistors will permanently resolve problem?

can someone draw schematics for Sonoff 4CH pro where exactly put resistors?
![image](https://user-images.githubusercontent.com/37035814/43514497-a4b8b5e2-9580-11e8-8cfe-8e578a97460d.png)

it would be nice to have complete guide, to help others with same issue

**Comment by Jason2866** [comment]
Yes, pullups will (probably) solve your problems. Sorry i dont have a 4ch pro, so cant guide you...

**Comment by mkomarac** [comment]
at which device have You had problems? 
Sonoff 4CH Pro?

I am asking because it has no sense at all, that channels 1, 2 and 4 working properly without issue or two in 6 months, and channel 3 is keep switching on and off constantly. It was the time that all has worked properly, over two months, (since HomeAssistant 056 until version 071) after upgrading to 073, 074 issues returned. I have reverted back to 071m but issue remained.
so, right now I have light show in my house, or some ghost switching, which reminds me on one episode of The Big Bang Theory about remote switching lights in students home in Beijing. So they said, that someone switching lights from Pasadena, CA. So funny. But I don't think that Leonard or Sheldon are switching my lights on and off.  I don't know what to do more. pull up resistors are probably solution, but why isn't reflecting to another switch in the same box... so stupid.

**Comment by ascillato** [comment]
Hi,

Have you managed to solve your issue?

**Comment by rPraml** [comment]
I have a similar issue,
always when the device crashes or reconnect to the mqtt broker (mosquitto 1.4.12) or even if I just restart mosquitto, the sonoff 1ch toggles.

It is 100% reproduceable, on every `sudo /etc/init.d/mosquitto restart` or device reboot, the switch toggles. I als tried to change the PowerRetain and LWT (SetOption10) value, but without success.
Any Ideas what can I try?


```
22:51:45 MQT: Connect failed to 10.75.14.1:1883, rc -2. Retry in 10 sec
22:51:57 MQT: Attempting connection...
22:51:57 MQT: Connected
22:51:57 MQT: tele/sw-schlafzimmer/LWT = Online (retained)
22:51:57 MQT: cmnd/sw-schlafzimmer/POWER = 
22:51:57 MQT: stat/sw-schlafzimmer/RESULT = {"POWER1":"OFF"}
22:51:57 MQT: stat/sw-schlafzimmer/POWER1 = OFF
```

mosquitto.log
```
1535579517: New connection from 10.75.14.82 on port 1883.
1535579517: New client connected from 10.75.14.82 as DVES_8E23BA (c1, k15, u'DVES_USER').
1535579517: Sending CONNACK to DVES_8E23BA (0, 0)
1535579517: Received PUBLISH from DVES_8E23BA (d0, q0, r1, m0, 'tele/sw-schlafzimmer/LWT', ... (6 bytes))
1535579517: Received PUBLISH from DVES_8E23BA (d0, q0, r0, m0, 'cmnd/sw-schlafzimmer/POWER', ... (0 bytes))
1535579517: Received SUBSCRIBE from DVES_8E23BA
1535579517:     cmnd/sw-schlafzimmer/# (QoS 0)
1535579517: DVES_8E23BA 0 cmnd/sw-schlafzimmer/#
1535579517: Sending SUBACK to DVES_8E23BA
1535579517: Sending PUBLISH to DVES_8E23BA (d0, q0, r1, m0, 'cmnd/sw-schlafzimmer/POWER', ... (6 bytes))
1535579517: Received SUBSCRIBE from DVES_8E23BA
1535579517:     cmnd/sonoffs/# (QoS 0)
1535579517: DVES_8E23BA 0 cmnd/sonoffs/#
1535579517: Sending SUBACK to DVES_8E23BA
1535579517: Received PUBLISH from DVES_8E23BA (d0, q0, r0, m0, 'stat/sw-schlafzimmer/RESULT', ... (16 bytes))
1535579517: Received PUBLISH from DVES_8E23BA (d0, q0, r0, m0, 'stat/sw-schlafzimmer/POWER1', ... (3 bytes))
1535579517: Received SUBSCRIBE from DVES_8E23BA
1535579517:     cmnd/DVES_8E23BA/# (QoS 0)
1535579517: DVES_8E23BA 0 cmnd/DVES_8E23BA/#
1535579517: Sending SUBACK to DVES_8E23BA
```

**Comment by ascillato** [comment]
@rPraml your case is not ghost switching. 

your log: `22:51:57 MQT: cmnd/sw-schlafzimmer/POWER =` means that Tasmota is receiving that command from your mqtt broker

You are using retain values in your broker.

Please, search in issues for examples on how to solve that.

**Comment by ascillato** [comment]
@mkomarac

Hi,

Have you managed to solve your issue?

**Comment by au190** [comment]
Hi have you resolved the problem ?

**Comment by ascillato2** [comment]
Hi,

Closing issue as there is no feedback. Please, ask to reopen if needed. Thanks

**Comment by mkomarac** [comment]
no
the problem is still on

**Comment by mkomarac** [comment]
After implementing R-C filter, everything worked great for 3 days.
after that time, problem reappears. but now on the different switch
![image](https://user-images.githubusercontent.com/37035814/44974189-30c87e80-af5f-11e8-84bc-7c24faeab560.png)
so the problem still exists.

more help please.

my lights are working like turn signal on car :(

**Comment by HobbytronicsPK** [comment]
I suspect, Tasmota uses button interrupts.

Whenever you will do the stuff via interrupt, you will face this issue even
after adding up RC-Filter.

If a heavy load in vicinity will be toggled, interrupt will detect a false
trigger.

Polling the button state in loop may resolve the issue.

**Comment by mkomarac** [comment]
6 LED lamps are per switch 3 and 4 (gu 5.3 led 12v , 2W)
so I don't believe there is too much load on relays

can You please explain, "Polling the button state in loop may resolve the issue."

tnx

**Comment by MightyMilster** [comment]
Hi, I also had ghost-switching. Tried adding a pull-up resistor. Replaced old sonoff with a new sonoff. Set Tasmota to "PowerRetain - On". None of this helped and the blinking continued.

I eventually solved my problem and thought I'd add it here in case it helps someone else going crazy like I was. Remember that we are dealing with controlling a high voltage device. When you add a manual switch to GPIO14, you are dealing with a low voltage. This low voltage is usually being carried on very thin wire. What happens when high voltage cables are placed near low voltage cables - Induction!!!! Mysterious unpredictable ghost switching. I suggest keeping the two as far apart as possible and definitely use shielded cable for your switch wires. Good luck!

**Comment by mkomarac** [comment]
I am using cat6 shielded cable for manual switching. if that is the case, why another switch works normal without ghost switching?

**Comment by MightyMilster** [comment]
This sort of interference can be VERY random and hard to pin-point. cat6 is twisted pairs and the 4 pairs are twisted too. Just takes one point in the cable to be sitting along side a high voltage cable for it to interfere. Can you run a test a test cable to the problem sonoff, which is nowhere near your high voltage cabling? If there are any pressure points where one is crushed/clamped against the other you could try placing folded aluminium foil between high and low voltage cables. Does the cat 6 go all the way through to the PCB? Look for any sections that are not shielded that could be suing from induction.

**Comment by mkomarac** [comment]
will do that, and report back

**Comment by mkomarac** [comment]
OK
after rearranging wires, I can  say that situation is getting better. but not quite. 
I have also configure tasmota according this video: https://www.youtube.com/watch?v=31IyfM1gygo
and indeed, the situation is getting better. but still, it's not reliable. 
the light still came on randomly, without any interaction.

for example, the most annoying situation, is at late night, just before going to sleep, while watching tv.
the eyes are wide open, adjusted to watching tv, and lights in living room came on. so shocking, annoying. 
I am about to give up, and throw this device away. 
Can't believe, that device is meant to be installed into power cabinet, rail mounted, and according to issues I am experiencing, issues are generated because is mounted into power cabinet.

I'll post picture, to show complete installation. 

is there any more thing I can try. any suggestions?

**Comment by au190** [comment]
Can you see át that time disconnect connect of that device in the mosquitto log ?

**Comment by au190** [comment]
Log
1537991642: Client BedRoom already connected, closing old connection.

1537991642: New client connected from 192.168.2.16 as BedRoom (c1, k15, u'xcc0').
1537992150: Saving in-memory database to /data/mosquitto.db.



I have mosquitto broker. If you go logs you shoud see something like this. Because the ghost switching can come if your Esp reconnects to the broker and retension is different on the Esp and hassio.

**Comment by kbaras** [comment]
Hello,

I have similar issues with one of my PRO's, and one of my DUAL's. But the problem comes up only when I shutdown the router (wifi signal is lost). Then the random switchings starting up.

**Comment by fnatalucci** [comment]
I have the same problem. I checked on Openhab2 log panel and found this

`2018-11-28 00:59:18.955 [ERROR] [.mqtt.internal.MqttMessageSubscriber] - Error processing MQTT message.

org.openhab.core.transform.TransformationException: An error occurred while opening file.

	at org.openhab.core.transform.TransformationHelper$TransformationServiceDelegate.transform(TransformationHelper.java:67) [215:org.openhab.core.compat1x:2.3.0]

	at org.openhab.binding.mqtt.internal.MqttMessageSubscriber.processMessage(MqttMessageSubscriber.java:138) [214:org.openhab.binding.mqtt:1.12.0]

	at org.openhab.io.transport.mqtt.internal.MqttBrokerConnection.messageArrived(MqttBrokerConnection.java:574) [216:org.openhab.io.transport.mqtt:1.12.0]

	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:475) [216:org.openhab.io.transport.mqtt:1.12.0]

	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:379) [216:org.openhab.io.transport.mqtt:1.12.0]

	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:183) [216:org.openhab.io.transport.mqtt:1.12.0]

	at java.lang.Thread.run(Thread.java:748) [?:?]`

**Comment by fnatalucci** [comment]
I have the same error on reboot random lights on

any solutions?

**Comment by ascillato** [comment]
use command `weblog 4` in your console and restart your device again to see where the power on comes from.

If it say SRC: MQTT you have retained messages in your broker. You need to use retain false in your broker
If it say SRC: SWITCH or BUTTON, you have ghost switching. If this is your case, please search in issues about that with the solutions proposed by other users.

**Comment by fnatalucci** [comment]
SRC is MQTT but retain mode is set to false

**Comment by ascillato** [comment]
Ok, you set to false, but the retained messages are still in the MQTT database. You need to delete your MQTT database.

**Comment by AdamWeglarz** [comment]
For me the issue is with physical momentary switch. I have uploaded AFE 1.3.1 FW to my sonoff 4ch pro and setup sensitivity time to 150ms and the problem seems to be gone (with 50ms the problem was similar to what I have observed on tasmota 6.3.0).
Is it possible to configure time for momentary switch to work somewhere in Tasmota?
Adam

**Comment by morenonew2** [comment]
I solved my issue !!! Finally!! For someday may TAsmota Sonoff 4chPro r2 POWER1 toggled on/off "apperently" randomly. I tried all solution found in Internet, but nothing solve my problem. Even I disconnect from MQTT server, even I change type of module configuration POWER 1 toggled on/off. But I find that when my alarm PIR sensor detected me the POWER1 switch ON and off. Bingo!!. I think that I had paired my POWER 1 switch button when my PIR detect me. So now in depaired my button switch and ALL IS OK

**Comment by mkomarac** [comment]
I have tried all suggestions from above, and one switch of 4CH pro is working unreliably. number 3. I have tried RC filters, wrapping wires that came IN and OUT from Sonoff into teflon wrap, rearranged wires, replace them, erase all RF codes from Sonoff, set retains, all suggested in all threads I could find, YT.

And still in Sonoff 4CHpro channel no3 keeps triggering itself. randomly. not as turning light, like before, but it's unreliable. So, I have disconnected it, and part of my lights in living room can't be powered ON. this is better than randomly turning ON. Especially late night, while watching TV before sleep, just to fall into, the lights came ON. aaaaa. I will accept any more suggestion, to resolve my problem, and help others as well.

**Comment by AdamWeglarz** [comment]
Hi,

I have migrated to AFE and disabled switch (you are not able to switch from button only from web or mqtt api). Solved all issues!

Adam

**Comment by mkomarac** [comment]
but I want to have option to use hard switches. I ran wires from push buttons directly to wall buttons (doorbell switch) using CAT6 cable so when I press that switch it simulates button press on the sonoff built into wall on the rail mount. wires are soldered directly on sonoff push buttons with RC filter, to avoid interferences. 
that's initial idea, replace simple wall switches, with doorbell buttons simulating Sonoff buttons presses.

did I explain good enough? :)

**Comment by djtonyd** [comment]
hi mkomarac 

i had the exact same issue as you. i currently have 5 sonoff's flashed and working on "Alexa & Smart-things"  1 out of the 5 sonoff's started playing up 6 months ago, it started turning on & off randomly at night and during the day. i would also turn it off manually and back on then off, it would seem to go away for awhile but happen 2 days later then yer. i have toggle switches and use the GPIO inputs and they are far from the 240vac wires, there wouldn't be much induction from that. my guess is poor manufacturing on the sonoff and how it was flashed. As i purchased another one and flashed it i  installed it again and          (All is good)         still working good till this day and thats over 6 months ago

**Comment by 2016for** [comment]
Man someone find a solution to the random switching please! :)

**Comment by ecirtaplem** [comment]
> Man someone find a solution to the random switching please! :)

Hello 2016for. 
I solved my problem by totally remove MQTT blocker (Mosquito) and reinstalled it. There was something remained in the Mosquito db and the only solution I found was that.

**Comment by mu2004** [comment]
I am using Sonoff T1 US 3 Gang, and the second switch always turns on by itself. What's interesting is, before turning on by itself, T1 always beeps a few times.

**Comment by djtonyd** [comment]
I’ve fixed mine with the 10phm resistor and cap trick so far so good 👍 even quicker response on the app unless it’s all in my head lol

**Comment by joecas65** [comment]
> I am using Sonoff T1 US 3 Gang, and the second switch always turns on by itself. What's interesting is, before turning on by itself, T1 always beeps a few times.

Same problem here with sonoff T1 US 1 Gang.
I hear 1 beep, then 2 beeps, then the T1 is on
Please someone find a solution...

**Comment by Nightreaver** [comment]
I just wondered why I wake up and my lights are on... first time I thought... I might have forgotton...
But right now, I heard some beeps... and my light turned on, Sonoff T1 3Gang, only 1st gang turned on this time, last time it was 3rd gang

**Comment by Nightreaver** [comment]
@djtonyd What kind of "Trick" is that? Providing some more details would be helpful. Thank you

**Comment by meingraham** [comment]
@Nightreaver 

Take a look at [this](https://github.com/arendst/Sonoff-Tasmota/wiki/FAQ#my-device-randomly-switches-on-and-off-do-i-have-ghosts-in-my-house).

Or try a [low pass filter](https://en.wikipedia.org/wiki/Low-pass_filter) 
![image](https://user-images.githubusercontent.com/34340210/57134422-266fb600-6d74-11e9-95de-b2785c30f7df.png)

**Comment by djtonyd** [comment]
Night reaver I thought I fixed em but 1 of my bank of downlights are still flickering really quick on / off then nothing for another 15/20mins

**Comment by meingraham** [comment]
@djtonyd Do you need a higher resistance?

https://www.youtube.com/watch?v=aq8_os6g13s

**Comment by djtonyd** [comment]
Yes I’ve already done all that and still occasionally does it on 1 bank of lights

**Comment by orientalsniper** [comment]
I'm also having random power-on and power-off, I can hear some beeps before it powers off.

**Comment by meingraham** [comment]
Have you given this a look - https://www.youtube.com/watch?v=31IyfM1gygo

**Comment by Nightreaver** [comment]
Another question is... why is it losing wifi signal at all? Its 1.5m from the wifi AP, if cant just lose wifi?! 
I got that retain setting thing, but it doesnt explain why its looking for retain settings at all.
Today I had 2 switches on a 3 gang, turning on one by one within 3 hours or so.

**Comment by meingraham** [comment]
Take a look at the [FAQ](https://github.com/arendst/Sonoff-Tasmota/wiki/FAQ) and [Troubleshooting](https://github.com/arendst/Sonoff-Tasmota/wiki/Troubleshooting) articles. The [wiki](https://github.com/arendst/Sonoff-Tasmota/wiki) is a gold mine of information!

Please, address any further questions to the [Tasmota Support Discord Chat](https://discord.gg/Ks2Kzd4). The chat is a better and more dynamic channel for helping you. Github issues are best used for Tasmota software feature requests and bug reporting. Troubleshooting is more effective using an interactive forum.

Please check the [Contributing Guideline and Policy](https://github.com/arendst/Sonoff-Tasmota/blob/development/CONTRIBUTING.md) and the [Support Guide](https://github.com/arendst/Sonoff-Tasmota/blob/development/SUPPORT.md).

Thanks.

### Support Information
See [Wiki](https://github.com/arendst/Sonoff-Tasmota/wiki) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
See [Community](https://groups.google.com/d/forum/sonoffusers) for forum.
See [Code of Conduct](https://github.com/arendst/Sonoff-Tasmota/blob/development/CODE_OF_CONDUCT.md)

**Comment by HobbytronicsPK** [comment]
Have a look on to [Ghost / Random Switching on Sonoff Devices (MQTT Retained)](https://blog.hobbytronics.pk/sonoff-tasmota-tips-tricks/#Ghost-Random-Switching)

**Comment by Nightreaver** [comment]
Well that didnt solve the issue for me, as i did this many times. my MQTT didnt even have persistance enabled...

What solved it for me was changing the CAP as mentioned in here - https://github.com/arendst/Sonoff-Tasmota/issues/5449

I currently use a 15nF+4,7nF cap so I'm at 19,7nF currently (with 22nF probably be the default) and my Ghost switching is gone while reaction is still fine. 

This might only help if changing the retain settings didnt help and your problem doesnt occur when wifi was lost or similar.

**Comment by ericalvim** [comment]
Same issue here in my Sonoff T1 3CH US.
Changed the settings to SwitchRetain Off; ButtonRetain On; ButtonRetain Off; PowerOnState 3; PowerRetain On
removed /var/lib/mosquitto/mosquitto.db, restarted the service, but the ghost switching still happens.

**Comment by alphaprawn** [comment]
Seems like people are seeing a fix with a low pass filter, but only if using the right cap. See night reavers last comment of 20 days ago In #2192.

**Comment by Korte68** [comment]
I solved this issue by changing Mosquitto broker to Home Assistant embedded broker...

**Comment by ascillato** [comment]
@Korte68 

That means that you had retained messages. A new broker will have a blank database. No retained messages. So, if you install again mosquitto, it will have also a blank database. Is up to you which broker to use. Both works really good.

**Comment by Korte68** [comment]
Tried to remove and install mosquitto...no luck. ghost switching again..

**Comment by Korte68** [comment]
maybe some mosquitto default parameter cause it...but I am not an mosquitto expert...

**Comment by ascillato** [comment]
>Tried to remove and install mosquitto...no luck. ghost switching again

So, that means that mosquitto database was not deleted. There are several tools for monitoring mqtt communications like hyveqs. With those you can see the retained messages. As soon as you connect to the broker, it sends all the retained messages at once, and you can delete them sending to the broker a blank payload to the retained topic.

**Comment by Korte68** [comment]
I run mosquitto on a synology NAS. Removed the package...and installed again.
Where should i check the mosquitto.db?

**Comment by ascillato** [comment]
I don't own a synology NAS, so I don't know, but as my comment before, you can delete the retained messages with any mqtt monitoring tool.

**Comment by luca4268** [comment]
Hi,
 I have some sonoff 4ch installed, 3 4ch r2 and 1 4ch pro r2. When those had up to 7.2, everything was fine. I moved all to the 8.1 some days ago and the 4ch pro r2 started behaving wrong with relay 2.
That relay just flips instantly at any time without reason. If I use the switch, the light will be driven correctly and,  it's my feeling, the issue seems to stay quite for sometime.
 
Such 4ch pro r2 is used to control lights, it's controlled by switches. it's connected to a mosquitto instance which reports the flipping coming from the sonoff itself.

The 4ch pro r2 console reports:
21:39:31 MQT: tele/lights/SENSOR = {"Time":"2020-01-14T21:39:31","Switch1":"ON","Switch2":"ON","Switch3":"OFF","Switch4":"OFF","DS18B20":{"Id":"000005B32B5A","Temperature":19.9},"TempUnit":"C"}
21:44:11 MQT: stat/lights/RESULT = {"POWER2":"ON"}
21:44:11 MQT: stat/lights/POWER2 = ON
21:44:11 MQT: stat/lights/RESULT = {"POWER2":"OFF"}
21:44:11 MQT: stat/lights/POWER2 = OFF

The mosquitto logs reports:
Tue Jan 14 21:44:12 2020: Received PUBLISH from lights (d0, q0, r0, m0, 'stat/lights/RESULT', ... (15 bytes))
Tue Jan 14 21:44:12 2020: Received PUBLISH from lights (d0, q0, r0, m0, 'stat/lights/POWER2', ... (2 bytes))
Tue Jan 14 21:44:12 2020: Sending PUBLISH to OpenHab (d0, q0, r0, m0, 'stat/lights/POWER2', ... (2 bytes))
Tue Jan 14 21:44:12 2020: Received PUBLISH from lights (d0, q0, r0, m0, 'stat/lights/RESULT', ... (16 bytes))
Tue Jan 14 21:44:12 2020: Received PUBLISH from lights (d0, q0, r0, m0, 'stat/lights/POWER2', ... (3 bytes))
Tue Jan 14 21:44:12 2020: Sending PUBLISH to OpenHab (d0, q0, r0, m0, 'stat/lights/POWER2', ... (3 bytes))

so it looks like the 4ch pro r2 is the source of the problem. The problem started after some days I moved to 8.1; with 7.2 never had any prob... If needed I can roll back to 7.2, but I could not find any instruction for that; maybe I have just to flash the v7.2, but I'd like to be really sure as it will be a pain in the back to unbrick this unmounting everyting from the place it is sitting now.

**Comment by ascillato** [comment]
Please, address this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4). The chat is a better and more dynamic channel for helping you. Github issues are meant for Tasmota Software Bug Reporting.

Please check the [Contributing Guideline and Policy](https://github.com/arendst/Sonoff-Tasmota/blob/development/CONTRIBUTING.md) and the [Support Guide](https://github.com/arendst/Sonoff-Tasmota/blob/development/SUPPORT.md).

Thanks.

---------------------------

### Support Information 
See [Wiki](https://tasmota.github.io/docs/#/) for more information.
See [FAQ](https://github.com/arendst/Sonoff-Tasmota/wiki/FAQ) for common questions/answers and [links](https://github.com/arendst/Sonoff-Tasmota/wiki/FAQ#i-cannot-find-an-answer-here) if none of your question is in the list
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
See [Community](https://groups.google.com/d/forum/sonoffusers) for forum.
See [Code of Conduct](https://github.com/arendst/Sonoff-Tasmota/blob/development/CODE_OF_CONDUCT.md)

**Comment by luca4268** [comment]
Hi Adrian,

I wrote to the chat but nobody replied. Anyway, I need a quick suggestion.
How to rollback from 8.1 to 7.2, just flashing the 7.2 or ?

KR
L


Mobile:+39 348 1580446


Il giorno mar 14 gen 2020 alle ore 22:30 Adrian Scillato <
notifications@github.com> ha scritto:

**Comment by ascillato** [comment]
Try again in the chat. If nobody replies may be volunteers are working or sleeping. 

About rolling back, it can brick your device. If you really want to do that, the safest is to erase all flash and flash by serial. This procedure of flashing is explained in the docs step by step.

**Comment by luca4268** [comment]
Hi Adrian,

first of all I apologise if I sounded pushing, not my intention. Again I
apologise :-).
Anyway I have another 4ch pro r2in v7.2. I'll swap them and I'll see if the
issue is still present having such running without anything connected and
in a different place.
If I meet that again I'll try to inform on it, maybe it's a very hidden
problem that for some reason I faced...
Thx again for your help

Ciao
Luca



thx a lot.
Mobile:+39 348 1580446


Il giorno gio 16 gen 2020 alle ore 16:27 Adrian Scillato <
notifications@github.com> ha scritto:

**Comment by ngrigoriev** [comment]
Brand new Sonoff 4CH Pro R2, flashed with latest 8.3.1. I have started to observe the ghost switching almost from the moment I flashed the device. Interestingly enough, it starts doing this only after I select the module configuration "Sonoff 4CH Pro". It does not seem to be happening if I keep it as "Sonoff Basic". Every ~5-10 minutes it would switch the relay 3 on and off, randomly. The device is sitting on the desk, powered via 110V, perfect WiFi connection, no MQTT configured yet, so clearly not an issue with the retained messages. I have attempted to reset the configuration and reconfigure it from scratch - same thing. Here is the log:

```
00:00:00 CFG: Loaded from flash at F6, Count 22
00:00:00 QPC: Flag 7E
00:00:00 CFG: CR 367/699
00:00:00 SRC: Restart
00:00:00 Project tasmota Tasmota Version 8.3.1(tasmota)-2_7_1
00:00:00 WIF: Checking connection...
00:00:00 WIF: Attempting connection...
00:00:00 WIF: Connecting to AP1 GHOSTWHEEL Channel 11 BSSId 02:4F:57:DC:19:F0 in mode 11N as tasmota_BDB80B-6155...
00:00:01 WIF: Checking connection...
00:00:01 WIF: Connected
00:00:01 HTP: Web server active on tasmota_BDB80B-6155 with IP address 10.10.10.101
00:00:03 RSL: tele/tasmota_BDB80B/INFO1 = {"Module":"Sonoff 4CH Pro","Version":"8.3.1(tasmota)","FallbackTopic":"cmnd/DVES_BDB80B_fb/","GroupTopic":"cmnd/tasmotas/"}
00:00:03 RSL: tele/tasmota_BDB80B/INFO2 = {"WebServerMode":"Admin","Hostname":"tasmota_BDB80B-6155","IPAddress":"10.10.10.101"}
00:00:03 RSL: tele/tasmota_BDB80B/INFO3 = {"RestartReason":"Software/System restart"}
00:00:03 RSL: stat/tasmota_BDB80B/RESULT = {"POWER1":"OFF"}
00:00:03 RSL: stat/tasmota_BDB80B/POWER1 = OFF
00:00:03 RSL: stat/tasmota_BDB80B/RESULT = {"POWER2":"OFF"}
00:00:03 RSL: stat/tasmota_BDB80B/POWER2 = OFF
00:00:03 RSL: stat/tasmota_BDB80B/RESULT = {"POWER3":"OFF"}
00:00:03 RSL: stat/tasmota_BDB80B/POWER3 = OFF
00:00:03 RSL: stat/tasmota_BDB80B/RESULT = {"POWER4":"OFF"}
00:00:03 RSL: stat/tasmota_BDB80B/POWER4 = OFF
00:00:06 HTP: Main Menu
00:00:06 QPC: Reset
00:00:07 RSL: tele/tasmota_BDB80B/STATE = {"Time":"1970-01-01T00:00:07","Uptime":"0T00:00:09","UptimeSec":9,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","Wifi":{"AP":1,"SSId":"GHOSTWHEEL","BSSId":"02:4F:57:DC:19:F0","Channel":11,"RSSI":100,"Signal":-34,"LinkCount":1,"Downtime":"0T00:00:03"}}
00:00:08 HTP: Console
00:00:08 APP: Boot Count 6
00:00:08 CFG: Saved to flash at F5, Count 23, Bytes 4096
00:00:21 WIF: Checking connection...
00:00:41 WIF: Checking connection...
00:00:52 HTP: Main Menu
00:00:54 HTP: Configuration
00:00:55 HTP: Configure Module
00:01:01 WIF: Checking connection...
00:01:21 WIF: Checking connection...
00:01:31 APP: Button3 level 1-0
00:01:31 SRC: Button
00:01:31 RSL: stat/tasmota_BDB80B/RESULT = {"POWER3":"ON"}
00:01:31 RSL: stat/tasmota_BDB80B/POWER3 = ON
00:01:31 APP: Button3 level 0-1
00:01:31 CFG: Saved to flash at F4, Count 24, Bytes 4096
00:01:41 WIF: Checking connection...
00:02:01 WIF: Checking connection...
00:02:21 WIF: Checking connection...
00:02:40 HTP: Configuration
00:02:41 WIF: Checking connection...
00:02:43 HTP: Main Menu
00:02:44 HTP: Console
```

Seems to be always the BUTTON3. I am not touching the physical buttons at all.

**Comment by Jason2866** [comment]
To avoid hunting maybe a ghost. Please change your setup so that the device gets the correct time.

**Comment by ngrigoriev** [comment]
Configured an NTP server, cleared the unused NTP slots. Still the same result - switches on and off randomly and frequently:

```
13:49:42 NTP: UTC 2020-06-02T12:49:41, DST 2020-03-29T02:00:00, STD 2020-10-25T03:00:00
13:49:42 RSL: INFO1 = {"Module":"Sonoff 4CH Pro","Version":"8.3.1(tasmota)","FallbackTopic":"cmnd/DVES_BDB80B_fb/","GroupTopic":"cmnd/tasmotas/"}
13:49:42 RSL: INFO2 = {"WebServerMode":"Admin","Hostname":"tasmota_BDB80B-6155","IPAddress":"10.10.10.101"}
13:49:42 RSL: INFO3 = {"RestartReason":"Power On"}
13:49:42 RSL: RESULT = {"POWER1":"OFF"}
13:49:42 RSL: POWER1 = OFF
13:49:42 RSL: RESULT = {"POWER2":"OFF"}
13:49:42 RSL: POWER2 = OFF
13:49:42 RSL: RESULT = {"POWER3":"OFF"}
13:49:42 RSL: POWER3 = OFF
13:49:42 RSL: RESULT = {"POWER4":"OFF"}
13:49:42 RSL: POWER4 = OFF
13:49:46 QPC: Reset
13:49:46 RSL: STATE = {"Time":"2020-06-02T13:49:46","Uptime":"0T00:00:08","UptimeSec":8,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","Wifi":{"AP":1,"SSId":"GHOSTWHEEL","BSSId":"02:4F:57:DC:19:F0","Channel":11,"RSSI":96,"Signal":-52,"LinkCount":1,"Downtime":"0T00:00:03"}}
13:49:48 APP: Boot Count 14
13:49:48 CFG: Saved to flash at FB, Count 105, Bytes 4096
13:49:50 HTP: Console
13:49:50 HTP: Console
13:49:54 APP: Button1 level 1-0
13:49:54 SRC: Button
13:49:54 RSL: RESULT = {"POWER1":"ON"}
13:49:54 RSL: POWER1 = ON
13:49:54 APP: Button1 level 0-1
13:49:55 CFG: Saved to flash at FA, Count 106, Bytes 4096
13:50:01 WIF: Checking connection...
13:50:21 WIF: Checking connection...
13:50:41 WIF: Checking connection...
13:50:51 APP: Button1 level 1-0
13:50:51 SRC: Button
13:50:51 RSL: RESULT = {"POWER1":"OFF"}
13:50:51 RSL: POWER1 = OFF
13:50:51 APP: Button1 level 0-1
13:50:52 APP: Button1 level 1-0
13:50:52 SRC: Button
13:50:52 RSL: RESULT = {"POWER1":"ON"}
13:50:52 RSL: POWER1 = ON
13:50:52 APP: Button1 level 0-1
13:50:52 APP: Button1 level 1-0
13:50:52 SRC: Button
13:50:52 RSL: RESULT = {"POWER1":"OFF"}
13:50:52 RSL: POWER1 = OFF
13:50:52 APP: Button1 level 0-1
13:50:52 APP: Button1 level 1-0
13:50:52 SRC: Button
13:50:52 RSL: RESULT = {"POWER1":"ON"}
13:50:52 RSL: POWER1 = ON
13:50:52 APP: Button1 level 0-1
```

**Comment by arendst** [comment]
Pls try to reset possibly learned RF signals.

**Comment by Jason2866** [comment]
Dip switches on the board in default position?

**Comment by ngrigoriev** [comment]
@arendst I think that was it.

@Jason2866 I have verified, I had S6 set to 0, all K5=1, all K6=0. I have powered off the board, flipped S6 to 1 (as per https://tasmota.github.io/docs/devices/Sonoff-4CH-Pro/#4ch-pro-dipswitch-configuration). Powered the board on. Then I did hold the "RF clear button" for about a minute after that and powered the board off/on to be sure.

It's been about 4 hours, I have left the board running with a small 7.5V power supply and then tried my 110V connection again. I have NOT noticed a single phantom switching happening! I have attempted to switch S6 back to 0 and let the board alone for a while, no changes, I no longer observe the unexpected switching. Console log is clear. So it appears that resetting the RF (assuming I have done it right ;) ) did make a difference. Another 433MHz device somewhere nearby, a garage door opener at the neighbour's house or something?

I am wondering how it would look like on the console if the button is triggered by an RF signal. I see that "SRC" log line logs the source and it is different for Web Console. What I had in my log before looks identical to what I get if I press the physical button now.

Thanks to everyone for help!!!

**Comment by SASpirit** [comment]
So replicating the issue is pretty simple.

**SETUP**: With the unit set to Inching mode, power-up device.  K5 to 0000, S6 to 1

Now each time you press one of the channel buttons it seems to change the persisted state of the button. i.e. I push channel one once (inching 1s), reboot/restart the the unit and channels 3 -4 actually power on during boot-up.  Press channel 1 again, reboot/restart the device and now channel 1 powers on during boot-up.  So each press of the buttons, 1- 4 seems to set a persisted state that is applied during boot-up causing phantom events to occur. 

I use this to open my garage door so as you can imagine on every alternating click if there is a power cycle or some form of reset even on my router it seems to trigger the switch last state causing my garage to open.

**Comment by ngrigoriev** [comment]
Mine just started doing it. Just like that - after working correctly for two years. Every minute, POWER4 OFF/ON, click-click. I was lucky to witness the beginning of this behavior, otherwise my pool heatpump would not like being turned off and on once a minute. Settings WebLog to 4 does show the source "Button". Tried to increase the value of Debounce to 1000, it seems that it starts happening less frequently after that but still happening every few minutes.

**Comment by ngrigoriev** [comment]
Did RF reset by pressing S5 button, no more ghost signals for 12+ hours so far. Well, it is bad that the issue can occur on its own even after not having it for a long time.

I wonder if there is a way to implement some kind of "equipment protection" logic in/with Tasmota? We know that the ghost switching exists and there is no reliable way to fix it. Frequent switching can damage the connected equipment. What would be great is to set the limit at firmware level, something like "if more then X button presses in Y seconds, block in state Z for W seconds" or something like this. Or also send MQTT message as an alert of possible malfunctioning.

**Comment by Jason2866** [comment]
Do a hardware modification. Disable the RF receiver

**Comment by ecirtaplem** [comment]
> Did RF reset by pressing S5 button, no more ghost signals for 12+ hours so far. Well, it is bad that the issue can occur on its own even after not having it for a long time.
> 
> I wonder if there is a way to implement some kind of "equipment protection" logic in/with Tasmota? We know that the ghost switching exists and there is no reliable way to fix it. Frequent switching can damage the connected equipment. What would be great is to set the limit at firmware level, something like "if more then X button presses in Y seconds, block in state Z for W seconds" or something like this. Or also send MQTT message as an alert of possible malfunctioning.

The only solution that worked for me is to uninstall MQTT broker and reinstall it. I kept the same setting for the broker so I had nothing to on all devices. Now no more Ghost at home.

**Comment by sfromis** [comment]
In that case, you've played fast at loose with retained MQTT messages. Do not set retain flags unless you know exactly why, and are prepared to deal with the trouble retained messages will very often cause when other things happen.

**Comment by ecirtaplem** [comment]
Yes @sfromis you are absolutely right, but I faced this issue three years ago and I was a newbie regarding Tasmota and MQTT and this is exactly the symptoms I got. 
Thanks to the HookUp I watched [this video](https://www.youtube.com/watch?v=31IyfM1gygo&t=11s) that solved my issue but I had also to reinstall MQTT broker.
So I hope this can help

**Comment by ngrigoriev** [comment]
@Jason2866 could you pls point me to the description how to disable the RF (433Mhz) receiver? If there is one, by any chance. I understand that a different anthenna is used for WiFi - the small one printed on the board.

As for the rest, it was not MQTT-related. Unless I am missing something. Because the problem was happening even with MQTT broker disabled, after Sonoff device was restarted and the log did show, as you can see above, the source as "Button".
